src/app/clusters/bindings/binding-cluster.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ *
+ *    Copyright (c) 2020-2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Binding Server Cluster
+ ***************************************************************************/
+
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/bindings/binding-cluster.h>
+#include <app/clusters/bindings/binding-table.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <clusters/Binding/Attributes.h>
+#include <clusters/Binding/Metadata.h>
+#include <clusters/Binding/Structs.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using TargetStructType         = Binding::Structs::TargetStruct::Type;
+using DecodableBindingListType = Binding::Attributes::Binding::TypeInfo::DecodableType;
+
+// TODO: add binding table to the persistent storage
+namespace {
+
+bool IsValidBinding(const EndpointId localEndpoint, const TargetStructType & entry)
+{
+    bool isValid = false;
+
+    // Entry has endpoint, node id and no group id
+    if (!entry.group.HasValue() && entry.endpoint.HasValue() && entry.node.HasValue())
+    {
+        if (entry.cluster.HasValue())
+        {
+            // Valid node/endpoint/cluster binding
+            ReadOnlyBufferBuilder<ClusterId> clientClusters;
+            InteractionModelEngine::GetInstance()->GetDataModelProvider()->ClientClusters(localEndpoint, clientClusters);
+            for (auto & client : clientClusters.TakeBuffer())
+            {
+                if (client == entry.cluster.Value())
+                {
+                    isValid = true;
+                }
+            }
+        }
+        else
+        {
+            // Valid node/endpoint (no cluster id) binding
+            isValid = true;
+        }
+    }
+    // Entry has group id and no endpoint and node id
+    else if (!entry.endpoint.HasValue() && !entry.node.HasValue() && entry.group.HasValue())
```

Comment:
just return abvove and then we can avoid an else after rteturn by:

```
// Entry has group id and no endpoint or node-id.
return !entry.endpoint.HasValue() && !entry.node.HasValue() && entry.group.HasValue();
```

Reply:
Done